cmd/testwrapper/testwrapper.go

```diff
@@ -135,19 +138,32 @@ func runTests(ctx context.Context, attempt int, pt *packageTests, otherArgs []st
 			// ignore
 		case "run":
 			resultMap[name] = &testAttempt{
-				name: name,
+				Name:         name,
+				AttemptCount: attempt,
 			}
 		case "skip", "pass", "fail":
-			resultMap[name].outcome = goOutput.Action
+			resultMap[name].Outcome = goOutput.Action
 			ch <- resultMap[name]
 		case "output":
 			if strings.TrimSpace(goOutput.Output) == flakytest.FlakyTestLogMessage {
-				resultMap[name].isMarkedFlaky = true
+				resultMap[name].IsMarkedFlaky = true
 			} else {
 				resultMap[name].logs.WriteString(goOutput.Output)
 			}
 		}
 	}
+	for _, result := range resultMap {
+		testAttemptJson, _ := json.Marshal(result)
+		file, err := os.OpenFile("test_attempts.json", os.O_APPEND|os.O_WRONLY, 0644)
```

Comment:
this is super expensive. you should pass in an optional writer where you write do.
This should be optional behavior which we can turn on with a flag in CI.

You are also opening the same file N times, once in every loop. Sending down a writer would help a lot